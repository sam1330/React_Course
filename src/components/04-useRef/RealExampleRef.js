import React, { useRef, useState } from "react";
import MultipleCustomHooks from "../03-examples/MultipleCustomHooks";

const RealExampleRef = () => {

  const [show, setShow] = useState(false);

  const inputRef = useRef();

  return (
    <>
      <h1>Real Example Ref</h1>
      
      {show && <MultipleCustomHooks />}
      
      <button
        className="btn btn-success mt-5"
        onClick={() => {
          setShow(!show);
        }}
      >
        Show/Hide
      </button>
    </>
  )
};

export default RealExampleRef;