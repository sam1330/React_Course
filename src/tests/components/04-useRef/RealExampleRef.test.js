import React from "react";

import { shallow } from "enzyme";

import RealExampleRef from "../../../components/04-useRef/RealExampleRef";

describe("<RealExampleRef /> tests", () => {

  const wrapper = shallow(<RealExampleRef />);

  
  test('should render <RealExampleRef /> component', () => {

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });

  test('should render <MultipleCustomHooks /> component', () => {

    wrapper.find('button').simulate('click');

    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

  });
});