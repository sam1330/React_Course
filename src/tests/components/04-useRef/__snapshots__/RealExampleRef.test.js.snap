// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RealExampleRef /> tests should render <RealExampleRef /> component 1`] = `
<Fragment>
  <h1>
    Real Example Ref
  </h1>
  <button
    className="btn btn-success mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;

exports[`<RealExampleRef /> tests should render component correctly 1`] = `
<Fragment>
  <h1>
    Real Example Ref
  </h1>
  <button
    className="btn btn-success mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
